weave/ops_primitives/_dict_utils.py

```diff
@@ -32,8 +32,15 @@ class MergeInputTypes(typing.TypedDict):
 def typeddict_merge_output_type(
     input_types: MergeInputTypes,
 ) -> typing.Union[types.TypedDict, types.UnionType, types.NoneType, types.UnknownType]:
-    self = input_types["self"]
-    other = input_types["other"]
+    self: types.Type = input_types["self"]
```

Comment:
do you have a sense as to why we need to do Const unwrapping here, but not in all output type cases?

Reply:
because of L45 (RHS)